src/components/LicensePicker/utils.js

```diff
@@ -0,0 +1,124 @@
+    if (obj.hasOwnProperty('noassertion')) return NOASSERTION
+    if (obj.license) return `${obj.license}${obj.plus ? '+' : ''}${obj.exception ? ` WITH ${obj.exception}` : ''}`
+    const left =
+      obj.left && obj.left.conjunction && obj.left.conjunction.toLowerCase() === 'or'
+        ? `(${this.stringify(obj.left)})`
+        : this.stringify(obj.left)
+    const right =
+      obj.right && obj.right.conjunction && obj.right.conjunction.toLowerCase() === 'or'
+        ? `(${this.stringify(obj.right)})`
+        : this.stringify(obj.right)
+    return left && `${left} ${!isNil(right) && obj.conjunction ? `${obj.conjunction.toUpperCase()} ${right}` : ''}`
+  }
+
+  /**
+   * Creates a new License Object in the specified path
+   * @param  {} obj original rules object
+   * @param  {} path where to create the new group
+   * @returns updated object rules
+   */
+  static createGroup(obj, path) {
+    if (path.length > 1 && obj.hasOwnProperty(path[0]) && obj[path[0]].hasOwnProperty(path[1])) {
+      obj[path[0]] = this.createGroup(obj[path[0]], path.slice(1))
+      return obj
+    }
+    if (path.length === 0)
+      return {
+        left: obj,
+        conjunction: 'and',
+        right: this.createGroupObject()
+      }
+    return {
+      ...obj,
+      [path[0]]: {
+        ...obj[path[0]],
+        right: this.createRuleObject(obj[path[0]].conjunction, obj[path[0]].right, this.createGroupObject())
+      }
+    }
+  }
+
+  /**
+   * Creates a new License Rules in the specified path
+   * @param  {} conjunction used conjuction to merge the new rules objects
+   * @param  {} obj current object rules
+   * @param  {} path where apply the change
+   * @returns updated object rules
+   */
+  static createRules(conjunction, obj, path) {
+    if (path.length > 1 && obj.hasOwnProperty(path[0]) && obj[path[0]].hasOwnProperty(path[1])) {
+      obj[path[0]] = this.createRules(conjunction, obj[path[0]], path.slice(1))
+      return obj
+    }
+    if (path.length === 0) return this.createRuleObject(conjunction, obj.left || obj, obj.right && obj.right)
+    const ruleConjunction = obj[path[0]] && obj[path[0]].conjunction ? obj.conjunction : conjunction
+    const left =
+      path[0] === 'left'
+        ? obj.left && obj.left.conjunction
+          ? this.createRuleObject(conjunction, obj.left.left, obj.left.right)
+          : obj.left
+        : obj.right && obj.right.conjunction
+          ? obj.left
+          : obj.conjunction !== conjunction
+            ? obj
+            : obj.left
+    const right =
+      path[0] === 'left'
+        ? obj.right
+        : obj.right && obj.right.conjunction
+          ? this.createRuleObject(conjunction, obj.right.left, obj.right.right)
+          : obj.conjunction === conjunction && this.createRuleObject(conjunction, obj.right)
+
+    return this.createRuleObject(ruleConjunction, left, right)
+  }
+
+  static createGroupObject() {
+    return {
+      left: { license: '' },
+      conjunction: 'and',
```

Comment:
why is it always 'and'?

Reply:
While adding a new group object, the user doesn't know yet what could be the joining conjunction of the left and right element. So, in order to create an empty object that could be parsed from the parse method, i've set a default 'and' conjunction.